routes/api.js

```diff
@@ -1,22 +1,62 @@
 const express = require( "express" );
 const router = express.Router();
+const bcrypt = require( "bcrypt" );
+const passport = require( "passport" );
+
+const passport_config = require( "../util/passportUtil" );
+const { checkNotAuthenticated } = passport_config;
+
+router.post( "/login", passport.authenticate( "local", {
+  successRedirect: "/home",
+  failureRedirect: "/login",
+  failureFlash   : true,
+} ) );
+
+router.post( "/register", checkNotAuthenticated, async ( req, res ) => {
+  try {
+    const user = await req.models.User.findOne(
+      {
+        where: { username: req.body.username },
+        raw  : true,
+      }
+    );
+    if ( user == null ) {
+      const hashedPassword = await bcrypt.hash( req.body.password, 10 );
```

Comment:
idk if already addressed, but deliberately no salting? Could also be done in the next PR, to justify its scope.

(It is a deliberate omission, as the [offical example is salted](https://www.npmjs.com/package/bcrypt#usage))

Reply:
There are more methods described in your link. The one I chose is a asynchronous one where only the original password and the rounds of salting are required.

Reply:
But I didn't even read the docs. You are salting already, it's just in the same callback :smile: